src/routing.js

```diff
@@ -208,17 +208,16 @@ export function RoutingProvider({ children }) {
   const getPathFromLocator = useCallback(
     locatorUpdate => {
       if (typeof locatorUpdate === 'function') {
-        locatorUpdate = locatorUpdate(locator)
+        locatorUpdate = locatorUpdate(locator) || {}
       }
 
       return getPath({
+        ...locator,
```

Comment:
The problem was that we weren't passing along the `locator.mode.orgAddress` above, causing this to not produce the correct path in the signer panel.

Wasn't sure deep merging was the best approach vs. individually selecting the keys we expect, but felt like the most safe in case we add more keys to `locator` in the future.

Reply:
Well found! I think this is fine for now.

Reply:
Right, as mentioned in https://github.com/aragon/aragon/pull/1457#discussion_r448048812, it sounds like we should prefer to use the function variant of `path()` when we want to extend an existing mode.

What do you think about staying as close to `React.setState()`'s behaviour as possible then—if you change an object, you overwrite that object entirely (to the point where we don't even inherit `mode.name`)?

Reply:
> What do you think about staying as close to `React.setState()`'s behaviour as possible then—if you change an object, you overwrite that object entirely (to the point where we don't even inherit `mode.name`)?

That sounds good yes. We could do this and also have errors?

- When `mode.name` is not set.
- When `mode.orgAddress` is not set (org mode).
- When `mode.instancePath` is set without `mode.instanceId` (org mode).
- When `mode.status` is not set (onboarding mode).
- When `preferences.subsection` is set without `preferences.section`.

Reply:
To be clear, we would use `log()` for the errors?

Reply:
I was thinking of throwing? To make sure we don’t forget anything when updating the mode.

Reply:
Closing an offline discussion, we decided to:

- Log a warning
- On production, send an error to sentry
  - Move the sentry configuration to be environment variable-controlled
  - Only initialize sentry on `*.aragon.org` domains (`SENTRY_DOMAIN=*.aragon.org`)

Reply:
Logged warnings in c88176d; will add the sentry bits in another PR.